### ISSUE TITLE
segfault on py_call_impl()

### ISSUE THREAD

**Issue by KevinLi-USDA** [rstudio/reticulate#1475]
```
> reticulate::py_config()
python:         /Users/kevinli/anaconda3/envs/invest-3130/bin/python
libpython:      /Users/kevinli/anaconda3/envs/invest-3130/lib/libpython3.9.dylib
pythonhome:     /Users/kevinli/anaconda3/envs/invest-3130:/Users/kevinli/anaconda3/envs/invest-3130
version:        3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 05:00:30)  [Clang 11.0.1 ]
numpy:          /Users/kevinli/anaconda3/envs/invest-3130/lib/python3.9/site-packages/numpy
numpy_version:  1.22.3

NOTE: Python version was forced by use_python() function
```
I am using `reticulate` to call the "seasonal water yield" module of the [natcap.invest](https://invest.readthedocs.io/en/latest/) Python package. I am running it in a conda virtual environment. The function takes a data dictionary to run (`args_swy`, which is loaded from "mod_args.py"). I think additional data files would be needed to reproduce this error exactly, but I don't know to what degree that is needed. Please let me know if any more specific information would be helpful. 

```
reticulate::use_condaenv(condaenv = "invest-3130")
reticulate::source_python("./python/mod_args.py")

swy <- reticulate::import("natcap.invest.seasonal_water_yield.seasonal_water_yield")

swy$execute(args_swy)
```
Results in:
```
 *** caught segfault ***
address 0x0, cause 'invalid permissions'

Traceback:
 1: py_call_impl(callable, call_args$unnamed, call_args$named)
 2: swy$execute(args_swy)

Possible actions:
1: abort (with core dump, if enabled)
2: normal R exit
3: exit R without saving workspace
4: exit R saving workspace
```

**Comment by t-kalinowski** [comment]
Are you able to reproduce this error outside of conda?
```r
virtualenv_create("invest-3130", version = "3.9", packages = c("natcap", ....))
```

Can you run R from gdb/lldb and provide a backtrace? 
```bash
R -d gdb
r # tell gdb to start R
reticulate::source_python(....) # reproduce segfault in R
bt # print backtrace
```

**Comment by KevinLi-USDA** [comment]
I am running into trouble creating a virtual environment with reticulate. Not sure if I'm doing right:

```
> virtualenv_create("invest-test", version = "3.9", packages = c("gdal", "natcap"))
Error in stop_no_virtualenv_starter(version) : 
  Suitable Python installation for creating a venv not found.
Requested version constraint: 3.9
Please install Python with one of following methods:
- https://www.python.org/downloads/
- reticulate::install_python(version = '<version>')
```

So I tried installing python with `install_python(version = "3.9")`. Then I ran:

```
virtualenv_create("invest-test", version = "3.9")
use_virtualenv("invest-test")

conda_install(envname = "invest-test", package = "gdal")
conda_install(envname = "invest-test", package = "natcap.invest")
```

I used `conda_install` to install `gdal` separately because `gdal` has to be installed first using conda, [per natcap.invest instructions](https://invest.readthedocs.io/en/3.13.0/installing.html). I couldn't get `natcap.invest` to install using `py_install()` so I tried running it with `conda_install()` as well. This last command results in the error:

```
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - natcap.invest -> python[version='>=3.10,<3.11.0a0|>=3.10,<3.11.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.8,<3.9.0a0',build=*_cpython]

Your python: python=3.11

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.



Error: one or more Python packages failed to install [error code 1]
```

This confuses me because I thought I was using the virtual environment `invest-test`, which should have Python 3.9 installed on it.

When I run `R -d gdb`, I get: `/usr/local/bin/R: line 299: exec: gdb: not found`

**Comment by KevinLi-USDA** [comment]
Thanks for going more in depth with this issue. I've followed your code to build a new virtual environment, which at first stopped the segfault issue. However, it led to an error about the latest gdal version that I don't know how to solve because it seems that previous versions of gdal are not available on homebrew. Can installations be done with conda?

Trying another module of natcap that I need to use, `reticulate::import("natcap.invest.sdr.sdr")`, led to an issue with the pandas version that I solved by reverting to pandas 2.0.3. This then resulted in the segfault issue again in R and an aborted session in RStudio.

When I run `lldb`, I am able to run all the modules to produce outputs, though I get warning messages. I am not familiar with lldb, so I don't understand why this is behaving differently.
